libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from typing import Optional
+
+from kotaemon.base import Document, Param
+
+from .base import BaseReader
+
+
+class AzureAIDocumentIntelligenceLoader(BaseReader):
+    """Utilize Azure AI Document Intelligence to parse document
+
+    As of April 24, the supported file formats are: pdf, jpeg/jpg, png, bmp, tiff,
+    heif, docx, xlsx, pptx and html.
+    """
+
+    _dependencies = ["azure-ai-documentintelligence"]
+
+    endpoint: str = Param("Endpoint of Azure AI Document Intelligence")
+    credential: str = Param("Credential of Azure AI Document Intelligence")
+    model: str = Param(
+        "prebuilt-layout",
+        help=(
+            "Model to use for document analysis. Default is prebuilt-layout. "
+            "As of April 24, you can view the supported models [here]"
+            "(https://learn.microsoft.com/en-us/azure/ai-services/"
+            "document-intelligence/concept-model-overview?view=doc-intel-4.0.0"
+            "#model-analysis-features)"
+        ),
+    )
+
+    @Param.auto(depends_on=["endpoint", "credential"])
+    def client_(self):
+        try:
+            from azure.ai.documentintelligence import DocumentIntelligenceClient
+            from azure.core.credentials import AzureKeyCredential
+        except ImportError:
+            raise ImportError("Please install azure-ai-documentintelligence")
+
+        return DocumentIntelligenceClient(
+            self.endpoint, AzureKeyCredential(self.credential)
+        )
+
+    def run(
+        self, file_path: str | Path, extra_info: Optional[dict] = None, **kwargs
+    ) -> list[Document]:
+        return self.load_data(Path(file_path), extra_info=extra_info, **kwargs)
+
+    def load_data(
+        self, file_path: Path, extra_info: Optional[dict] = None, **kwargs
+    ) -> list[Document]:
+        metadata = extra_info or {}
+        with open(file_path, "rb") as fi:
+            poller = self.client_.begin_analyze_document(
+                self.model,
+                analyze_request=fi,
+                content_type="application/octet-stream",
+                output_content_format="markdown",
+            )
+            result = poller.result()
+
+        return [Document(content=result.content, metadata=metadata)]
```

Comment:
Even though Markdown output from Azure API directly is nice, we should note that it should be coupled with SemanticChunking to keep table and section structure in place. Reference: https://python.langchain.com/docs/modules/data_connection/document_transformers/markdown_header_metadata/

Reply:
Also, please note that figure image extraction only available in JSON output mode. In markdown mode we still get <figure> to specify figure region but only text or OCR content embedded inside.
![image](https://github.com/Cinnamon/kotaemon/assets/15842122/309c238c-0e8b-4ae8-8926-0d3c2a61347e)

Reply:
Alrighty, this should be considered as improvement, as we are still working on the best agnostic representation of these information for a file @taprosoft

Reply:
Sure, let add an issue for this @trducng.